tests/xharness/Jenkins/TestTasks/AggregatedRunSimulatorTask.cs

```diff
@@ -63,7 +63,11 @@ protected override async Task ExecuteAsync ()
 					await task.SelectSimulatorAsync ();
 				}
 
-				var devices = executingTasks.First ().Simulators;
+				var devices = executingTasks.FirstOrDefault ()?.Simulators; 
+				if (devices == null) { 
+					ExecutionResult = TestExecutingResult.Ignored;
```

Comment:
No simulator means the test should fail with TestExecutingResult.DeviceNotFound.

Reply:
@rolfbjarne fixed.